### DISCUSSION TITLE
Copy/paste multiple layers?

### DISCUSSION THREAD

**Discussion by Jboten** [keiwando/toonsquid#522] [Q&A] [answered]
Is there a way to duplicate or copy/paste multiple layers at once? I’m trying to do a frame by frame music video of a band of skeletons and each skeleton has on average 20 different bones as separate layers (each skeleton is set up with its own group folder), all hierarchied. As it stands I seem to only be able to copy/paste one layer at a time, so with 20 layers for each of my four skeletons, that’s 80 copy/paste actions for each frame.

**Comment by Cat-Butler** [comment]
Possibly stupid question: Why not just duplicate the overall skeleton folder on the timeline, after one skeleton is complete, thus including a duplicate of all hierarchied sub-layers? Animate any shared keyframes (e.g. nodding heads/skulls) before duplication, then add differing keyframes (guitar versus piano) after duplication.

Idle thought: To keep the keyframes tidy, you might also want to either add a Transform layer to the timeline and make the skeleton group a hierarchy child of it, or add a Transform layer into the Group, and make all the pieces into hierarchy children of the Transform layer (group layers propagate/copy all transform keyframes into each sub-layer, transform layers keep the keyframes on the transform layer). Use that Transform layer to place/move each skeleton as a whole.

**Reply by Jboten** [reply]
Not a stupid question at all, and your response is appreciated! Unfortunately, unless I’m misunderstanding, I don’t think it accomplishes what I’m after necessarily. In my case I’ve already got the multiple skeletons in separate group folders for each one, separately rigged for each of their component bones. Each skeleton is performing in significantly different motion so there wouldn’t be a lot of shared key frames, and I’m trying to avoid tweening animations anyway, to give it a more raw old animation feel. I’m animating on twos so every two frames is a whole new drawing frame for each layer in the group. What I’m trying to do is copy the previous two frames, paste them, then adjust drawing as necessary for whatever given piece is being animated at that point. I COULD extend the drawing frame on layers until that a given bone needs to move, but that gets a little more messy in my mind I guess. I’m attaching a screenshot to maybe make it clearer what I’m doing. 

![IMG_0094](https://github.com/keiwando/toonsquid/assets/133608249/45e9db22-c879-4886-9c1e-03a7726df610)

**Reply by Cat-Butler** [reply]
Hmm, I'm not sure that ToonSquid is quite ready for that particular workflow. 

Possible weak workarounds:

1) It is possible to animate at 12 fps and export at 24 fps and ToonSquid will extrapolate frames, effectively doubling them.
That avoids the two-frame issue, and you can just hammer on Duplicate for each layer. Some video apps might also convert 12 fps to 24 fps without a speed change (DaVinci Resolve does on desktop, not sure if it does on iPad).

2) As you've said, you could extend a Drawing, and just Duplicate off the end of it when you want to make individually repainted layer changes. Although if you have it rigged by hierarchy, would you need to change the Drawing content or just keyframe it (rotation, mainly, perhaps some scaling) for some of the bones? For subtle distortions, I suppose that modification is important.

3) I'm a big fan of rigging, so I'd tend to use symbols and frame markers e.g. Eyes symbol, then keyframe between states: Normal, Surprised, etc,  or Hand symbol and keyframe between Open, Closed, etc, but then you do lose the ability to make more subtle and sophisticated changes and distortions. Of course, there's nothing stopping you from copying the contents of the last frame in the symbol range and modifying it afterwards in a new frame after the symbol range, similar to an extended Drawing.

In the end, I guess animation (and especially frame-by-frame animation) is still a lot of work, which is why so many animation studios go for team-based workflows.

Sorry, personally I cannot think of any quick and definite solutions for this particular case. Maybe somebody else will have a brilliant solution.

**Comment by andydjames** [comment]
If layers are grouped, I think you can copy the group on the timeline and then paste it including all child layers. (I think)

**Reply by Jboten** [reply]
You’re right, but that pastes a whole new folder and new rigged skeleton. What I’m looking for is copy/pasting the individual layers respective frames within the group. I posted a screenshot on another commenters response that might explain it a little better. Thank you for your reply though!

**Reply by andydjames** [reply]
If you lose the keyframe info, I think after pasting the layers (in a group or individually), you can copy and paste each layer's keyframe data. Good luck!

**Comment by keiwando** [comment, marked as answer]
With the 1.10.7 update the "Add drawing" and "Duplicate" buttons in the timeline now also work with multiple selected layers.
In your screenshot above:
1. go to frame 3
2. multiselect all of the animation layers with the different skeleton parts (e.g. by dragging a multiselection rectangle from frame 3 towards the left and selecting the drawings, which also selects their respective layers)
3. Tap the "Duplicate" button in the timeline

That should give you the result you were looking for.

**Reply by Jboten** [reply]
I must have been behind an update. I tried that and it only duplicated the uppermost animation layer of my multi selection. For sanity i checked for an update and sure enough there was one available. Updated and tried again, worked like a charm! Thanks so much for your help and this amazing app!

**Reply by keiwando** [reply]
You weren't behind when you posted this. I was already working on a bugfix update when I read your post and figured that I might as well also make this relatively small adjustment to these buttons while I'm at it. Happy to hear that you like the app!
